### DISCUSSION TITLE
Statistics in violin plots

### DISCUSSION THREAD

**Discussion by FernandaScar** [satijalab/seurat#5997] [Q&A]
Hello, everyone!

I'm new at Single Cell, sorry if it is a basic question.
I'm using a public data (available on GEO - **GSE93374**), using the DGE.txt data to perform vlnplots with my genes of interest. I wanted to add p value into the plot and also I wanted to compare between three conditions.

I created a seurat object from DGE.txt file (seurat), and to plot the Vlnplot I'm using the following commands (using ggplot2 and ggpubr libraries):

```
##make a function 
vp_case1 <- function(gene_signature, file_name, test_sign, y_max){
  plot_case1 <- function(signature){
    VlnPlot(seurat, features = signature,
            pt.size = 0.1, 
           group.by = "orig.ident", 
            y.max = y_max,
    ) + stat_compare_means(comparisons = test_sign, label = "p.signif")
  }
  purrr::map(gene_signature, plot_case1) %>% cowplot::plot_grid(plotlist = .)
  file_name <- paste0(file_name, "_r.png")
  ggsave(file_name, width = 14, height = 8)
}

##make the vlnplot
gene_sig <- c("Hspd1", "Lonp1")
comparisons <- list(c("Chow", "Fasted", "HFD"))
vp_case1(gene_signature = gene_sig, file_name = "Desktop/gene_sig", test_sign = comparisons, y_max = 8) +
state_compare_means(method = "anova")

```

However, my problems with this commands are: I'm getting the comparison with all conditions not only chow, fasted and HFD samples; and also, I'm getting the p value from kruskal-wallis method, not from anova.

Does anyone know how can I work with this?

I'm using this links as a reference: https://www.biostars.org/p/458261/ and http://www.sthda.com/english/articles/24-ggpubr-publication-ready-plots/76-add-p-values-and-significance-levels-to-ggplots/

**Comment by samuel-marsh** [comment]
Hi,

I would not recommend doing statistics off of the Violin plot data using normal statistical tests (i.e. t-test or anova via ggpubr).  Statistics of changes between genes across groups should be performed using differential gene expression analysis and proper DE methods (i.e. `FindMarkers` or pseudobulk methods using DESe2 or EdgeR etc).

Best,
Sam

**Reply by FernandaScar** [reply]
Hi Sam,

Thank you for your answer! I'm not sure how I can use DE methods to compare changes between groups with specific genes, most of the tutorials and scripts that I found used FindMarkers() or FindAllMarkers() to find the markers between clusters and compare this markers with the other clusters, not informing the genes that they want to compare.

There is any way that I can get this information with DE methods (using DGE data)?
Sorry if this question doesn't apply, I'm still learning what I can do with this data. :)

**Reply by samuel-marsh** [reply]
Hi,

So FindMarkers gets the differences between what is supplied to ident.1, ident.2, and subset.ident depending on group.by.

Here is example using example meta data column.
```
pbmc <- pbmc3k.SeuratData::pbmc3k.final

pbmc$sample_id <- sample(c("sample1", "sample2", "sample3", "sample4", "sample5", "sample6"), size = ncol(pbmc), replace = TRUE)

# Get list of markers different between sample1 and the other samples within the cluster named "B"
markers <- FindMarkers(pbmc, group.by = "sample_id", ident.1 = "sample1", subset.ident = "B")
```


Alternatively if you were just comparing within one cluster and had subset that cluster you could just change what the active ident is:
```
pbmc <- pbmc3k.SeuratData::pbmc3k.final

pbmc$sample_id <- sample(c("sample1", "sample2", "sample3", "sample4", "sample5", "sample6"), size = ncol(pbmc), replace = TRUE)

pbmc_b <- subset(pbmc, idents = "B")

Idents(pbmc) <- "sample_id"

markers <- FindMarkers(pbmc, ident.1 = "sample1")
```

Best,
Sam
